### ISSUE TITLE
Add Istio Stack

### ISSUE THREAD

**Issue by kalantar** [iter8-tools/iter8-install#50]
**Is your feature request related to a problem? Please describe.**
Be able to install an Istio stack.

**Describe the solution you'd like**

- [ ] RBAC rules allowing controller/handler access to VirtualServices and DestinationRules
- [ ] Sample Prometheus based metrics
- [ ] ServiceMonitor for Istio dataplane

**Describe alternatives you've considered**

**Acceptance criteria**
Iter8 should install, tests should pass

**Comment by sriumcp** [comment]
[ ] Remove rbac rules for helm and [ ] Remove sample specific rbac rules (for helm) are duplicates?

I agree with the need for doing this, but I think this needs to be a separate issue, and needs to be done carefully.

In particular, doing this without doing updating the helm sample will cause helm sample to fail.

**Comment by kalantar** [comment]
I agree that it requires changes to the helm example. But I don't think the changes here cause that problem since that example is currently gated by the earlier version of the install code, isn't it?

I expect to create a second issue on iter8 to document the changes that will be needed given the changes to install

**Comment by kalantar** [comment]
Created issue https://github.com/iter8-tools/iter8/issues/615

**Comment by sriumcp** [comment]
Ok. Sounds good. In this case, it would be preferable if we upgrade the current samples first, so that they work on this install, and then bring in the Istio samples in a subsequent PR, instead of bundling them all together. This was anyway, my main motivation for suggesting the separation of Helm issue (to keep PRs short and self-contained).

**Comment by kalantar** [comment]
This issue included additional restructuring work. A new issue, https://github.com/iter8-tools/iter8-install/issues/54, was created for that work. This issue is restricted to the Istio stack.
